numpy/_core/src/multiarray/ctors.c

```diff
@@ -2233,7 +2233,8 @@ PyArray_FromInterface(PyObject *origin)
                 }
             }
         }
-        Py_DECREF(descr);
```

Comment:
Why not just move the `Py_DECREF(descr)` up inside the branch where `descr` is defined? Outside of that branch, it is guaranteed to be `NULL` (similarly, the definition of `new_dtype` can be much closer to its use...).

Reply:
Now looked at the original issue, where I see @seberg also noted it should be put up: reason is that if `GetStringRef` returns 0, then the key was not present so `descr` cannot have been set.